### PR TITLE
Fix some fast up to date check issues in VS

### PR DESCRIPTION
**Pull request by Pilchie**
Fix several issues:

1. Places where it's explicitly disabled.
2. Places where we used CopyToOutputDirectory="Always"
3. One project wasn't actually set to be built in VS.
4. A few things marked as input or output files that don't exist and aren't generated.

There are still 3 projects that don't build properly, and 5 that fail the fast up to date check for me with this, but getting closer.

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @KirillOsenkov, just because I always do when messing with FastUpToDateCheck, and @drewnoakes because there is one case I still don't understand.

Remaining issues:

```
172>FastUpToDate: Output 'C:\Code\aspnetcore\artifacts\obj\Microsoft.AspNetCore.App.Runtime\Debug\net6.0\win-x64\Microsoft.AspNetCore.App.Runtime.dll' does not exist, not up to date. (Microsoft.AspNetCore.App.Runtime)

410>FastUpToDate: No inputs are newer than earliest output 'C:\Code\aspnetcore\artifacts\bin\dotnet-microsoft.openapi.Tests\Debug\net6.0\Microsoft.DotNet.Open.Api.Tools.Tests.pdb' (4/24/2021 3:53:21 PM). (dotnet-microsoft.openapi.Tests)
410>FastUpToDate: Latest write timestamp on input marker is 4/24/2021 3:55:08 PM on 'C:\Code\aspnetcore\artifacts\obj\Microsoft.dotnet-openapi\Debug\net6.0\Microsoft.dotnet-openapi.csproj.CopyComplete'. (dotnet-microsoft.openapi.Tests)
410>FastUpToDate: Write timestamp on output marker is 4/24/2021 3:55:08 PM on 'C:\Code\aspnetcore\artifacts\obj\dotnet-microsoft.openapi.Tests\Debug\net6.0\dotnet-microsoft.openapi.Tests.csproj.CopyComplete'. (dotnet-microsoft.openapi.Tests)
410>FastUpToDate: Input marker is newer than output marker, not up to date. (dotnet-microsoft.openapi.Tests)

413>FastUpToDate: Output 'C:\Code\aspnetcore\artifacts\bin\Microsoft.Extensions.ApiDescription.Server\Debug\netcoreapp2.1\Microsoft.Extensions.ApiDescription.Server.dll' does not exist, not up to date. (Microsoft.Extensions.ApiDescription.Server)

458>FastUpToDate: Output 'C:\Code\aspnetcore\artifacts\obj\Microsoft.AspNetCore.ANCMSymbols\Debug\net6.0\win-x64\Microsoft.AspNetCore.ANCMSymbols.dll' does not exist, not up to date. (Microsoft.AspNetCore.ANCMSymbols)

494>FastUpToDate: Output 'C:\Code\aspnetcore\artifacts\obj\Microsoft.AspNetCore.App.Ref\Debug\net6.0\Microsoft.AspNetCore.App.Ref.dll' does not exist, not up to date. (Microsoft.AspNetCore.App.Ref)
```

172, 413, 458, and 494 are projects that don't actually generate an output assembly.  Any idea how to make them play nicer with the fast up to date check @dotnet/aspnet-build or @drewnoakes.

410 is the one that is puzzling me. I don't actually know what that log means.  @drewnoakes can you help me out?

File: AspNetCore.sln [hunk 1]
```diff
@@ -6353,11 +6353,13 @@ Global
 		{F8AB0D33-1082-4F41-90E7-77C06EDD983D}.Release|x86.ActiveCfg = Release|Any CPU
 		{F8AB0D33-1082-4F41-90E7-77C06EDD983D}.Release|x86.Build.0 = Release|Any CPU
 		{647F0202-165F-4E98-A798-B2B3113050EC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{647F0202-165F-4E98-A798-B2B3113050EC}.Debug|Any CPU.Build.0 = Debug|Any CPU
```
Comment (Pilchie) [inline AspNetCore.sln:6356]: The lack of these lines meant this project (IISSamples) wasn't actually marked to build.

File: src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj [hunk 2]
```diff
@@ -32,7 +32,6 @@
   <ItemGroup>
     <YarnInputs Include="$(YarnWorkingDir)**" Exclude="$(YarnWorkingDir)node_modules\**;$(YarnWorkingDir)*.d.ts;$(YarnWorkingDir)dist\**" />
     <YarnOutputs Include="$(YarnWorkingDir)dist\$(Configuration)\AuthenticationService.js" />
-    <YarnOutputs Include="$(YarnWorkingDir)dist\$(Configuration)\AuthenticationService.js.map" Condition="'$(Configuration)' == 'Debug'" />
```
Comment (Pilchie) [inline src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj:35]: This doesn't seem to be being generated, but `YarnOutputs` is explicitly added to the `UpToDateCheckOutput` below.

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Microsoft.AspNetCore.Components.WebAssembly.Authentication.csproj [hunk 3]
```diff
@@ -33,7 +33,6 @@
   <ItemGroup>
     <YarnInputs Include="$(YarnWorkingDir)**" Exclude="$(YarnWorkingDir)node_modules\**;$(YarnWorkingDir)*.d.ts;$(YarnWorkingDir)dist\**" />
     <YarnOutputs Include="$(YarnWorkingDir)dist\$(Configuration)\AuthenticationService.js" />
-    <YarnOutputs Include="$(YarnWorkingDir)dist\$(Configuration)\AuthenticationService.js.map" Condition="'$(Configuration)' == 'Debug'" />
```
Comment (Pilchie) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/Microsoft.AspNetCore.Components.WebAssembly.Authentication.csproj:36]: This doesn't seem to be being generated, but `YarnOutputs` is explicitly added to the `UpToDateCheckOutput` below.

File: src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj [hunk 4]
```diff
@@ -7,7 +7,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore" />
     <Reference Include="Microsoft.AspNetCore.TestHost" />
-    <Content Include="Microsoft.AspNetCore.TestHost.StaticWebAssets.xml" CopyToOutputDirectory="Always" />
+    <Content Include="Microsoft.AspNetCore.TestHost.StaticWebAssets.xml" CopyToOutputDirectory="PreserveNewest" />
```
Comment (Pilchie) [inline src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj:10]: Having `Always` means this project (and anything that depends on it) will *always* rebuild in VS.

File: src/Servers/IIS/IIS/test/IIS.FunctionalTests/IIS.FunctionalTests.csproj [hunk 5]
```diff
@@ -3,7 +3,6 @@
   <PropertyGroup>
     <TargetFramework>$(DefaultNetCoreTargetFramework)</TargetFramework>
     <TestGroupName>IIS.FunctionalTests</TestGroupName>
-    <DisableFastUpToDateCheck>True</DisableFastUpToDateCheck>
```
Comment (Pilchie) [inline src/Servers/IIS/IIS/test/IIS.FunctionalTests/IIS.FunctionalTests.csproj:6]: A bunch of the IIS stuff seems to explicitly disable the FastUpToDateCheck. Any idea why @jkotalik?

File: src/SignalR/clients/ts/FunctionalTests/SignalR.Client.FunctionalTestApp.csproj [hunk 6]
```diff
@@ -10,8 +10,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <None Remove="selenium\run-ci-tests.ts" />
```
Comment (Pilchie) [inline src/SignalR/clients/ts/FunctionalTests/SignalR.Client.FunctionalTestApp.csproj:13]: These files don't exist on my machine.  Any ideas @BrennanConroy?

Comment (BrennanConroy) [reply src/SignalR/clients/ts/FunctionalTests/SignalR.Client.FunctionalTestApp.csproj:13]: Old test files I believe

**Comment by KirillOsenkov** [comment]
If you're looking at the build, I randomly noticed there are a couple of pending changes after a clean build (file BOM changes?) to blazor.server.js and a couple launchSettings.json files in tests rejiggered.

**Comment by KirillOsenkov** [comment]
Regarding 410, if you have a binlog, look at the timing of file writes for the CopyComplete markers - why is the CopyComplete marker for the Tests project written earlier than the marker for the non-test project? Could be insufficient build ordering?? (just random guesses here)

**Comment by KirillOsenkov** [comment]
You can append `$time` to the binlog viewer search query and it will show timestamps for results. So e.g. `$task Csc $time`. Also you can hover over targets, tasks etc. to view timestamps. There's also a crude Timeline view which I never have time to finish, but it can still be useful to understand the timing and ordering (time flows down, columns are build nodes).

**Comment by KirillOsenkov** [comment]
Hmm I've just realized it's the VS build, not command-line, it's harder to get a decent binlog that involves both projects. Not sure how to investigate this. If you keep building, does it still rebuild every time?

File: src/Servers/IIS/IIS/test/IIS.FunctionalTests/IIS.FunctionalTests.csproj (hunk 5 above)
Comment (halter73) [reply src/Servers/IIS/IIS/test/IIS.FunctionalTests/IIS.FunctionalTests.csproj:6]: Context: https://github.com/aspnet/IISIntegration/pull/1186

We should verify that VS will rebuild for native changes now.

**Comment by drewnoakes** [comment]
> 172, 413, 458, and 494 are projects that don't actually generate an output assembly

You can remove the output item from the build, however if that leaves no outputs then the fast up-to-date check will always build it.

> 410 is the one that is puzzling me. I don't actually know what that log means.
> ```
> 410>FastUpToDate: No inputs are newer than earliest output 'C:\Code\aspnetcore\artifacts\bin\dotnet-microsoft.openapi.Tests\Debug\net6.0\Microsoft.DotNet.Open.Api.Tools.Tests.pdb' (4/24/2021 3:53:21 PM). (dotnet-microsoft.openapi.Tests)
> 410>FastUpToDate: Latest write timestamp on input marker is 4/24/2021 3:55:08 PM on 'C:\Code\aspnetcore\artifacts\obj\Microsoft.dotnet-openapi\Debug\net6.0\Microsoft.dotnet-openapi.csproj.CopyComplete'. (dotnet-microsoft.openapi.Tests)
> 410>FastUpToDate: Write timestamp on output marker is 4/24/2021 3:55:08 PM on 'C:\Code\aspnetcore\artifacts\obj\dotnet-microsoft.openapi.Tests\Debug\net6.0\dotnet-microsoft.openapi.Tests.csproj.CopyComplete'. (dotnet-microsoft.openapi.Tests)
> 410>FastUpToDate: Input marker is newer than output marker, not up to date. (dotnet-microsoft.openapi.Tests)
> ```

It looks like the dependency from `Microsoft.dotnet-openapi.Tests.csproj` on `Microsoft.dotnet-openapi.csproj` isn't tracking properly. I'm not familiar with the `CopyComplete` files. Could it possibly be related to using a variable in the `ProjectReference` in `Microsoft.dotnet-openapi.Tests.csproj`? Seems unlikely. I tried building your fork but don't have all the right bits on my machine and don't have time today to dig into that. Is this a repeatable failure (is there an observable race)?

**Comment by drewnoakes** [comment]
There seems to be a bunch of special handling of `CopyComplete` for Razor projects. Does that apply here? For example:

https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.Compilation.targets#L272-L274

**Comment by Pilchie** [comment]
@drewnoakes it did seem to be consistent.

@pranavkm / @captainsafia for the potential Razor SDK impact.

I feel like we struggled with projects that produced no output assembly other places too.  @jasonmalinowski do you have any ideas there?  Does Roslyn do anything interesting?

**Comment by jasonmalinowski** [comment]
Roslyn has a few projects we are really using for packaging; right now those build time. I haven't looked at fighting with it, maybe @tmat might know details.

**Comment by captainsafia** [comment]
> There seems to be a bunch of special handling of CopyComplete for Razor projects. Does that apply here? For example:

That particular change only affects .NET 5 and below projects where we still produced a Views assembly. I don't suspect this will come into play with the changes here since we are targetting net6.0 and above.

**Review by dougbu — APPROVED** [review]

File: AspNetCore.sln [hunk 7]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.30204.135
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31223.326
```
[the remark this answers is not in the document]
Comment (Pilchie) [reply AspNetCore.sln:4]: I *think* it's fine as long as the `MinimumVisualStudioVersion` below isn't updated.

File: src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj (hunk 2 above)
Comment (Pilchie) [reply src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj:35]: Yeah, that's what I figured.  I wondered if anyone who knew yarn things knew whether it matters and if so, how to turn it back on instead.

File: src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj (hunk 4 above)
Comment (Pilchie) [reply src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj:10]: When you change branches, git will write the new source file, and it's timestamp will be newer, so the next build should still copy it.

File: src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj (hunk 2 above)
Comment (BrennanConroy) [reply src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj:35]: @javiercn

File: src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj (hunk 4 above)
Comment (drewnoakes) [reply src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj:10]: I would be very interested in knowing of a legitimate use case for `Always` as I'm not aware of one. I occasionally wonder if we could just treat it as an alias for `PreserveNewest`. `Always` is a foot gun, and we make it too easy for people to select it.

**Comment by Pilchie** [comment]
@BrennanConroy / @HaoK if this passes, any objection to merging?  There are still a few projects that don't produce outputs where the fast up to date check fails, but @drewnoakes is looking at those.

There are also two Blazor identity projects with issues, but I'll ask @javiercn to look at them when he's back.

Regardless, this set of changes should reduce needless rebuilds in VS significantly.

**Review by BrennanConroy — APPROVED** [review]
Looks good to me
Would be nice if NativeProjectReference handled the UpToDateCheckInput for us though 😃

**Comment by Pilchie** [comment]
> Would be nice if NativeProjectReference handled the UpToDateCheckInput for us though

Agreed, but @rainersigwald said there isn't a reliable way to go from vcxproj -> output.  Otherwise, we might be able to do it in our `ResolveNativeProjectReference` target.  (Though I'm not sure if the fast up to date check runs targets to collect `UpToDateCheckInput`, or if they have to be static).

**Review by HaoK — APPROVED** [review]

**Comment by rainersigwald** [comment]
> Otherwise, we might be able to do it in our `ResolveNativeProjectReference` target. (Though I'm not sure if the fast up to date check runs targets to collect `UpToDateCheckInput`, or if they have to be static).

I though they had to be evaluation time constant but that might be wrong: https://github.com/dotnet/project-system/blob/c954a632f74afdac9118a1f505257107789ff00e/docs/up-to-date-check.md#customization talks about a target. So it might be doable to add the NativeContent item to UpToDateCheckInput after RAR/design-time RAR. @drewnoakes, that might be worth clarifying in that doc.

File: src/Identity/ApiAuthorization.IdentityServer/test/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.Tests.csproj [hunk 8]
```diff
@@ -18,16 +18,16 @@
 
   <ItemGroup>
     <None Update="current.pfx">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (dougbu) [inline src/Identity/ApiAuthorization.IdentityServer/test/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.Tests.csproj:21]: Could you switch to attributes while changing this file❔ Each `Update` would fit on one line then. May even be able to fit all for updates on a single line 😺 

Might want to remove `CopyToOutputDirectory="{blah}"` entirely here too. @rainersigwald is that the default for `@(Content)` items❔

File: src/Servers/IIS/AspNetCoreModuleV2/Symbols/Microsoft.AspNetCore.ANCMSymbols.csproj [hunk 9]
```diff
@@ -30,6 +30,9 @@
   <ItemGroup Condition=" '$(BuildIisNativeProjects)' == 'true' ">
     <NativeProjectReference Include="$(MSBuildThisFileDirectory)..\..\AspNetCoreModuleV2\OutOfProcessRequestHandler\OutOfProcessRequestHandler.vcxproj" Platform="$(NativeVCPlatform)" />
     <NativeProjectReference Include="$(MSBuildThisFileDirectory)..\..\AspNetCoreModuleV2\AspNetCore\AspNetCore.vcxproj" Platform="$(NativeVCPlatform)" />
+
+    <UpToDateCheckInput Include="$(AspNetCoreModuleV2OutOfProcessHandlerDll)" />
+    <UpToDateCheckInput Include="$(AspNetCoreModuleV2ShimDll)" />
```
Comment (dougbu) [inline src/Servers/IIS/AspNetCoreModuleV2/Symbols/Microsoft.AspNetCore.ANCMSymbols.csproj:35]: Would it help to set `$(GeneratePackageOnBuild)` to `true` and add an `@(UpToDateCheckBuilt)` item for the produced package❔

**Comment by dougbu** [comment]
I'm confused by "Note that `UpToDateCheckOutput` exists but is deprecated and only maintained for backwards compatability." in the docs. The recommendation is to use `UpToDateCheckBuilt` instead but that controls project outputs, not inputs. @drewnoakes @rainersigwald❔

(Someone should fix the "compatibility" spelling in the doc too 😃)

**Comment by drewnoakes** [comment]
> I though they had to be evaluation time constant but that might be wrong: https://github.com/dotnet/project-system/blob/c954a632f74afdac9118a1f505257107789ff00e/docs/up-to-date-check.md#customization talks about a target. So it might be doable to add the NativeContent item to UpToDateCheckInput after RAR/design-time RAR. @drewnoakes, that might be worth clarifying in that doc.

@rainersigwald yes, you can modify the set of `UpToDateCheckInput/Output/Built` items via the targets specified in those docs. We use design-time build data for that.

I'm unclear on the clarification you're after though.

> I'm confused by "Note that UpToDateCheckOutput exists but is deprecated and only maintained for backwards compatability." in the docs.

@dougbu I know you edited this, but regardless: you're right, this is confusing. I've removed mention of the deprecation in https://github.com/dotnet/project-system/pull/7840. Technically `UpToDateCheckOutput` is redundant, but in practice it's less confusing to allow its use, and we can't remove it anyway.

> Hmm I've just realized it's the VS build, not command-line, it's harder to get a decent binlog that involves both projects. Not sure how to investigate this. If you keep building, does it still rebuild every time?

@KirillOsenkov since then you pointed me to the use of `MSBuildDebugEngine` environment variable for full-fidelity binlogs, as described in https://github.com/dotnet/project-system-tools#getting-higher-fidelity-logs-from-vs

@Pilchie can you suggest a solution filter and project to test the no-output case with? I've burnt my budget today on trying to get the solution building locally. Narrowing to a subset of projects might be helpful.

**Comment by Pilchie** [comment]
> Can you suggest a solution filter and project to test the no-output case with?

`src/Framework/Framework.slnf` is probably the best example.

File: src/Identity/ApiAuthorization.IdentityServer/test/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.Tests.csproj (hunk 8 above)
Comment (Pilchie) [reply src/Identity/ApiAuthorization.IdentityServer/test/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.Tests.csproj:21]: These aren't `Content`, they are `None`, so seems like we couldn't remove these?

File: src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj (hunk 2 above)
Comment (Pilchie) [reply src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj:37]: @dougbu - but the point of this PR is that people *do* build Debug locally in VS, and this makes their experience worse when doing so.

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Microsoft.AspNetCore.Components.WebAssembly.Authentication.csproj (hunk 3 above)
Comment (Pilchie) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/Microsoft.AspNetCore.Components.WebAssembly.Authentication.csproj:38]: As above - people *do* build it locally, and the file isn't generated there either.

File: src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj (hunk 4 above)
Comment (Pilchie) [reply src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj:10]: @rainersigwald pointed to https://github.com/dotnet/msbuild/blob/14efa06d736aabb3af01ff827fff9426518f5fac/src/Tasks/Microsoft.Common.CurrentVersion.targets#L745-L748.  They default to `Never`, so these can't be removed.

File: src/Identity/ApiAuthorization.IdentityServer/test/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.Tests.csproj (hunk 8 above)
Comment (Pilchie) [reply src/Identity/ApiAuthorization.IdentityServer/test/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.Tests.csproj]: Make them single line though

File: src/Servers/IIS/IIS/test/IIS.FunctionalTests/IIS.FunctionalTests.csproj (hunk 5 above)
Comment (Pilchie) [reply src/Servers/IIS/IIS/test/IIS.FunctionalTests/IIS.FunctionalTests.csproj:7]: Fixed by adding some things to `UpToDateCheckInput`.

File: src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj (hunk 2 above)
Comment (dougbu) [reply src/Components/WebAssembly/Authentication.Msal/src/Microsoft.Authentication.WebAssembly.Msal.csproj:37]: Ah, I misunderstood the situation 👍

File: src/Servers/IIS/AspNetCoreModuleV2/Symbols/Microsoft.AspNetCore.ANCMSymbols.csproj (hunk 9 above)
Comment (Pilchie) [reply src/Servers/IIS/AspNetCoreModuleV2/Symbols/Microsoft.AspNetCore.ANCMSymbols.csproj:35]: Are you proposing that to solve the fact that this project doesn't generate an output assembly, or for some other reason?

File: src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj (hunk 4 above)
Comment (dougbu) [reply src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/Microsoft.AspNetCore.Tests.csproj:10]: Agreed. However, we override that default in CSharp.Common.props for all C# test projects.

That said, I thought we did that for all projects. As-is, the override just creates inconsistencies. We should probably remove it at some point and _add_ missing `CopyToOutputDirectory` and `CopyToPublishDirectory` attributes. Something for another day&hellip;

File: src/Servers/IIS/AspNetCoreModuleV2/Symbols/Microsoft.AspNetCore.ANCMSymbols.csproj (hunk 9 above)
Comment (dougbu) [reply src/Servers/IIS/AspNetCoreModuleV2/Symbols/Microsoft.AspNetCore.ANCMSymbols.csproj:35]: Yes, to solve the lack of an output assembly -- if it works 😃

**Comment by Pilchie** [comment]
>  So it might be doable to add the NativeContent item to UpToDateCheckInput

Is the idea here to do something like:

```xml
<Target Name="CollectUpToDateCheckInputDesignTime" AfterTargets="ResolveAssemblyReference">
  <UpToDateCheckInput Include="@(NativeContent)" />
</Target>
```

And that would solve our native references?

@drewnoakes / @rainersigwald - does that look right?

**Comment by drewnoakes** [comment]
That's the idea. Check the implementation of that target in the .NET project system and make sure it's returning the same thing. From memory (I'm on a phone) you need `Returns="@(UpToDateCheckInput)"` on the target too. You may find `BeforeTargets` is easier.

I guess our documentation can be improved here with an example. I'll do that when I'm back.

**Comment by Pilchie** [comment]
Unable to repro the two npm/yarn projects writing to their own inputs today 🤷🏻‍♂️ .

**Comment by Pilchie** [comment]
Still looking if I can do something better with native content and no output projects, but I'm going to merge this in the meantime.

**Pull request merged**